### ISSUE TITLE
How to use `PiFutex::lock_pi_until()` with correct clock? 

### ISSUE THREAD

**Issue by rtzoeller** [m-ou-se/linux-futex#3]
I'm trying to use `PiFutex::lock_pi_until()` and am hitting an issue with `FUTEX_LOCK_PI` and `std::time::Instant` using different clocks. It's manifesting as `PiFutex::lock_pi_until()` returning nearly immediately, regardless of the timeout value.

For example,

```rust
let futex = linux_futex::PiFutex::<linux_futex::Private>::new(1); // Non-zero -> locked by someone else
println!("{:?}", std::time::Instant::now());
let result = futex.lock_pi_until(std::time::Instant::now() + std::time::Duration::from_secs(2));
println!("{:?}", std::time::Instant::now());
dbg!(result);
```

outputs

```
Instant { tv_sec: 12663, tv_nsec: 14076982 }
Instant { tv_sec: 12663, tv_nsec: 14100659 }
[tests/tests.rs:144] result = Err(
    TimedOut,
)
```

---

`FUTEX_LOCK_PI` is documented as using `CLOCK_REALTIME`, while `std::time::Instant` (used by `lock_pi_until()`) is implemented with `CLOCK_MONOTONIC`.

1. How is `PiFutex::lock_pi_until()` intended to be used? Is there a recommendation for converting between `std::time::SystemTime` (which uses `CLOCK_REALTIME`) and `std::time::Instant`?
2. Have you considered adding support for `FUTEX_LOCK_PI2`, which allows selecting a clock and could be used with `CLOCK_MONOTONIC`?

Thanks!

**Comment by rtzoeller** [comment]
It seems like modifying linux-futex to use `FUTEX_LOCK_PI2` instead "just works", but there's a much higher Linux kernel requirement associated with that. I'll at least add it to rust-lang/libc ([PR](https://github.com/rust-lang/libc/pull/2874/)), since it seems to be absent.

**Comment by m-ou-se** [comment]
Thanks! It's fixed in https://github.com/m-ou-se/linux-futex/commit/8f75c589a24bf81e78cfa0b3bd0186d7965ffdff. `lock_pi_until` now accepts either an SystemTime or an Instant, and picks either FUTEX_LOCK_PI or FUTEX_LOCK_PI2 depending on that.

Published as 0.1.2 to crates.io just now.

**Comment by rtzoeller** [comment]
@m-ou-se thanks! This new release seems to fix it.
